cli/internal/otel/opentelemetry.go

```diff
@@ -0,0 +1,185 @@
+package otel
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"time"
+
+	"github.com/rs/zerolog"
+
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/exporters/otlp/otlplog/otlploghttp"
+	otellog "go.opentelemetry.io/otel/log"
+	logglobal "go.opentelemetry.io/otel/log/global"
+	"go.opentelemetry.io/otel/sdk/log"
+	"go.opentelemetry.io/otel/sdk/resource"
+	semconv "go.opentelemetry.io/otel/semconv/v1.26.0"
+)
+
+// newResource returns a resource describing this application.
+func newResource() *resource.Resource {
+	def := resource.Default()
+	schemaURL := def.SchemaURL()
+	if schemaURL == "" {
+		schemaURL = semconv.SchemaURL
+	}
+	r, err := resource.Merge(
+		def,
+		resource.NewWithAttributes(
+			schemaURL,
+			semconv.ServiceName("cloudquery-cli"),
+		),
+	)
+	if err != nil {
+		panic(err)
+	}
+	return r
+}
+
+type otelConfig struct {
+	endpoint string
+}
+
+func getLogsProcessor(ctx context.Context, opts otelConfig) (*log.BatchProcessor, error) {
+	if opts.endpoint == "" {
+		return nil, nil
+	}
+
+	logOptions := []otlploghttp.Option{
+		otlploghttp.WithEndpoint(opts.endpoint),
+		otlploghttp.WithInsecure(),
+		otlploghttp.WithCompression(otlploghttp.GzipCompression),
+	}
+
+	exporter, err := otlploghttp.New(ctx, logOptions...)
+	if err != nil {
+		return nil, fmt.Errorf("creating OTLP log exporter: %w", err)
+	}
+
+	processor := log.NewBatchProcessor(exporter)
+	return processor, nil
+}
+
+func SetupOtel(ctx context.Context, logger zerolog.Logger, otelEndpoint string) (shutdown func(), err error) {
+	opts := otelConfig{
+		endpoint: otelEndpoint,
+	}
+
+	logsProcessor, err := getLogsProcessor(ctx, opts)
+	if err != nil {
+		return nil, err
+	}
+
+	pluginResource := newResource()
+	lp := log.NewLoggerProvider(
+		log.WithProcessor(logsProcessor),
+		log.WithResource(pluginResource),
+	)
+
+	otel.SetErrorHandler(otel.ErrorHandlerFunc(func(err error) {
+		logger.Warn().Err(err).Msg("otel error")
+	}))
+
+	logglobal.SetLoggerProvider(lp)
+
+	shutdown = func() {
+		if err := lp.Shutdown(context.Background()); err != nil {
+			logger.Error().Err(err).Msg("failed to shutdown OTLP logger provider")
+		}
+	}
+
+	return shutdown, nil
+}
+
+type otelLoggerHook struct {
+	otellog.Logger
+	ctx          context.Context
+	invocationId string
+}
+
+func (h *otelLoggerHook) Run(e *zerolog.Event, level zerolog.Level, message string) {
+	record := otellog.Record{}
+	record.SetTimestamp(time.Now().UTC())
+	record.SetSeverity(otellogSeverity(level))
+	record.SetBody(otellog.StringValue(message))
+	// See https://github.com/rs/zerolog/issues/493, this is ugly but it works
```

Comment:
I know this is how we do it in the plugin SDK, but there's a better way https://github.com/rs/zerolog/pull/682#issuecomment-2395677680, just I haven't had the time to update the SDK go.

However I'm good with this is @murarustefaan is good with it